src/code/shrink_window.c

```diff
@@ -1,79 +1,89 @@
 #include "global.h"
 
-s32 D_8012CED0 = 0;
+typedef enum {
+    /* 0 */ LETTERBOX_STATE_IDLE,
+    /* 1 */ LETTERBOX_STATE_GROWING,
+    /* 2 */ LETTERBOX_STATE_SHRINKING
+} LetterboxState;
 
-s32 sShrinkWindowVal = 0;
-s32 sShrinkWindowCurrentVal = 0;
+s32 sLetterboxState = LETTERBOX_STATE_IDLE;
 
-void ShrinkWindow_SetVal(s32 value) {
+s32 sLetterboxSizeTarget = 0;
+s32 sLetterboxSize = 0;
+
+void Letterbox_SetSizeTarget(s32 target) {
     if (HREG(80) == 0x13 && HREG(81) == 1) {
-        osSyncPrintf("shrink_window_setval(%d)\n", value);
+        osSyncPrintf("shrink_window_setval(%d)\n", target);
     }
-    sShrinkWindowVal = value;
+
+    sLetterboxSizeTarget = target;
 }
 
-u32 ShrinkWindow_GetVal(void) {
-    return sShrinkWindowVal;
+u32 Letterbox_GetSizeTarget(void) {
+    return sLetterboxSizeTarget;
 }
 
-void ShrinkWindow_SetCurrentVal(s32 currentVal) {
+void Letterbox_SetSize(s32 size) {
     if (HREG(80) == 0x13 && HREG(81) == 1) {
-        osSyncPrintf("shrink_window_setnowval(%d)\n", currentVal);
+        osSyncPrintf("shrink_window_setnowval(%d)\n", size);
     }
-    sShrinkWindowCurrentVal = currentVal;
+
+    sLetterboxSize = size;
 }
 
-u32 ShrinkWindow_GetCurrentVal(void) {
-    return sShrinkWindowCurrentVal;
+u32 Letterbox_GetSize(void) {
+    return sLetterboxSize;
 }
 
-void ShrinkWindow_Init(void) {
+void Letterbox_Init(void) {
     if (HREG(80) == 0x13 && HREG(81) == 1) {
         osSyncPrintf("shrink_window_init()\n");
     }
-    D_8012CED0 = 0;
-    sShrinkWindowVal = 0;
-    sShrinkWindowCurrentVal = 0;
+
+    sLetterboxState = LETTERBOX_STATE_IDLE;
+    sLetterboxSizeTarget = 0;
+    sLetterboxSize = 0;
 }
 
-void ShrinkWindow_Destroy(void) {
+void Letterbox_Destroy(void) {
     if (HREG(80) == 0x13 && HREG(81) == 1) {
         osSyncPrintf("shrink_window_cleanup()\n");
     }
-    sShrinkWindowCurrentVal = 0;
+
+    sLetterboxSize = 0;
 }
 
-void ShrinkWindow_Update(s32 updateRate) {
-    s32 off;
+void Letterbox_Update(s32 updateRate) {
+    s32 step;
 
     if (updateRate == 3) {
-        off = 10;
+        step = 10;
     } else {
-        off = 30 / updateRate;
+        step = 30 / updateRate;
     }
```

Comment:
Isn't the logic here backwards? 

| updateRate | FPS | Step |
|:----------:|:---:|:----:|
|      1     |  60 |  30  |
|      2     |  30 |  15  |
|      3     |  20 |  10  |

If I'm understanding this right, having a faster target update results in a bigger step value, meaning sLetterboxSizeTarget will be reached much quicker in real time at higher FPS instead of taking approximately constant time.

Reply:
you are correct though, at update rate 1 the bars appear pretty much instantly. which makes sense, a step is 30 and the target is 32